File: src/frontend/rate-my-fit/src/types/review.ts
import { UserCredentials, UserPreview } from './user';

export type Review = {
  fitID: string,
  grade: number,
  date: string,
  comment: string | null,
  reviewID: string,
  authorToken: string,
  author: UserPreview
};

export type Form = {
  fitID: string,
  grade: number,
  date: string,
  comment: string | null,
  userCredentials: UserCredentials
};

File: src/backend/models/review.py
from typing import Optional

from models.abstract import AbstractModel
from models.user import UserCredentialsModel


class ReviewModel(AbstractModel):
    userCredentials: UserCredentialsModel
    reviewID: Optional[str] = None
    fitID: str
    grade: int
    date: str
    comment: Optional[str] = None
